tests/containers/podmansh.pm

```diff
@@ -0,0 +1,191 @@
+# SUSE's openQA tests
+#
+# Copyright 2021-2025 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Test podman login shell.
+# - checks if podmansh runs and shell is indeed confined in the container
+# - validates login through sudo, su, ssh and tty prompt
+# Maintainer: QE-C team <qa-c@suse.de>
+
+use Mojo::Base 'containers::basetest';
+use testapi;
+use serial_terminal qw(select_serial_terminal select_user_serial_terminal);
+use Utils::Systemd qw(systemctl);
+use utils;
+use containers::common;
+
+my $src_image = "registry.opensuse.org/opensuse/leap";
+my $quadlet_container = <<_EOF_;
+[Unit]
+Description=Podman shell container
+After=local-fs.target
+
+[Container]
+Image=$src_image
+ContainerName=podmansh
+UserNS=keep-id
+RunInit=yes
+DropCapability=all
+NoNewPrivileges=true
+
+# change homedir to match user console serial terminal regexes
+Environment=HOME=/home
+WorkingDir=/home
+Environment=SHELL=/bin/bash
+
+# OpenQA serial terminal access
+AddDevice=/dev/ttyS0
+
+Exec=sleep infinity
+
+[Quadlet]
+# avoid infinite waiting for network by podman-user-wait-network-online.service
+# https://github.com/systemd/systemd/issues/28762
+DefaultDependencies=false
+
+[Install]
+RequiredBy=default.target
+_EOF_
+
+my $unit_name = 'podmansh.container';
+my $quadlet_dir = '/etc/containers/systemd';
+my $uid;
+my $systemd_user_path;
+
+my $initial_user_shell;
+
+my $podman;
+
+sub run {
+    my ($self) = @_;
+    select_serial_terminal;
+
+    $podman = $self->containers_factory('podman');
+    install_packages('podmansh');
+
+    # prepare normal user for rootless containers
+    prepare_user_account();
+
+    # enable user linger - required for login using su or sudo
+    # su/sudo does not create a systemd login session, which rootless podman depends on
+    # https://github.com/containers/podman/issues/17202#issuecomment-1402604841
+    assert_script_run("loginctl enable-linger $uid");
+    script_retry("loginctl list-users | grep '$username.*lingering'", retry => 5, delay => 5, timeout => 10);
+
+    # pre-pull the image
+    script_retry("sudo -su $username sh -c 'cd; podman pull $src_image'", retry => 3, delay => 60, timeout => 180);
```

Comment:
```suggestion
    script_retry("sudo -su $username podman pull $src_image", retry => 3, delay => 60, timeout => 180);
```

We don't need `cd` here. :)

Reply:
Oh but we do. :smile: 

Without this, the user shell would still be under `/root` and any operation would fail with permission denied, so we need to get back to the user's home directory.

Reply:
> Oh but we do. 😄
> 
> Without this, the user shell would still be under `/root` and any operation would fail with permission denied, so we need to get back to the user's home directory.

Why not use use sudo's `-D` option then?

Reply:
:thinking: Will try that.

Reply:
@ricardobranco777, I am getting this with t he `-D` option:

`sudo: you are not permitted to use the -D option with /bin/bash`

Login shell is also not an option for me because that is already handled by `podmansh`.